guide/qip/qip-processor.rst

```diff
@@ -101,47 +181,222 @@ To let it find the optimal pulses, we need to give the parameters for :func:`qut
+
+In the simulation, noise can be added to the processor at different levels:
 
 * The decoherence time T1 and T2 can be defined for the processor or for each qubit. When calculating the evolution, the corresponding collapse operators will be added automatically to the solver.
 
 * The noise of the physical parameters (e.g. detuned frequency) can be simulated by changing the parameters in the model, e.g. laser frequency in cavity QED. (This can only be time-independent since QuTiP open system solver only allows varying coefficients, not varying Hamiltonian operators.)
 
 * The noise of the pulse intensity can be simulated by modifying the coefficients of the Hamiltonian operators or even adding new Hamiltonians.
 
-To add noise to a processor, one needs to first define a noise object :class:`qutip.qip.noise.Noise`. The simplest relaxation noise can be defined directly in the processor with relaxation time. Other pre-defined noise can be found as subclasses of  :class:`qutip.qip.noise.Noise`. We add noise to the simulator with the method :meth:`qutip.qip.device.Processor.add_noise`.
+To add noise to a processor, one needs to first define a noise object :class:`~qutip.qip.noise.Noise`. The simplest relaxation noise can be defined directly in the processor with relaxation time. Other pre-defined noise can be found as subclasses of  :class:`~qutip.qip.noise.Noise`. We can add noise to the simulator with the method :meth:`~qutip.qip.device.Processor.add_noise`.
 
-Workflow of the Processor
-================================
-This section helps you understand the workflow inside the simulator.
+Below, we show two examples.
 
-.. image:: /figures/qip/processor-workflow.png
+The first example is a processor with one qubit under rotation around the z-axis and relaxation time :math:`T_2=5`. We measure the population of the :math:`\left| + \right\rangle` state and observe the Ramsey signal:
 
-The figure above shows how the noise is processed in ``processor``. The noise is defined separately in a class object. When called, it takes parameters and the unitary noiseless :class:`qutip.QobjEvo` from the processor, generates the noisy version and sends the noisy :class:`qutip.QobjEvo` together with the collapse operators to the processor.
+.. plot::
+
+    import numpy as np
+    import matplotlib.pyplot as plt
+    from qutip.qip.device import Processor
+    from qutip.operators import sigmaz, destroy
+    from qutip.qip.operations import snot
+    from qutip.states import basis
```

Comment:
I don't mind what you do with `qutip.qip` imports, but in core we don't guarantee that `sigmaz` and `destroy` will stay in the same physical place (and they do move in 5.0) - their logical location is top-level `qutip` only.  Similar for `basis`.

Reply:
Oh, really? That's a good point.

I'm also thinking for `qutip_qip` if we should import everything to the main domain or not... I'm going to open another discussion.